### PR TITLE
Add config id to heap dump for faster identification.

### PR DESCRIPTION
**Pull request by baldersheim**
@bjorncs PR

### CODE REVIEW

File: container-disc/src/main/java/com/yahoo/container/jdisc/ConfiguredApplication.java [hunk 1]
```diff
@@ -407,14 +408,21 @@ public void destroy() {
...
+
     // Workaround for ApplicationLoader.stop not being able to shutdown
     private void startShutdownDeadlineExecutor() {
         shutdownDeadlineExecutor = new ScheduledThreadPoolExecutor(1, new DaemonThreadFactory("Shutdown deadline timer"));
         shutdownDeadlineExecutor.setExecuteExistingDelayedTasksAfterShutdownPolicy(false);
         long delayMillis = (long)(shudownTimeoutS.get() * 1000.0);
         shutdownDeadlineExecutor.schedule(() -> {
             if (dumpHeapOnShutdownTimeout.get()) {
-                String heapDumpName = Defaults.getDefaults().underVespaHome("var/crash/java_pid.") + ProcessHandle.current().pid() + ".hprof";
+                String saneConfigId = configId.replaceAll(SANITIZE_FILENAME, "_");
```
Comment (bjorncs) [inline container-disc/src/main/java/com/yahoo/container/jdisc/ConfiguredApplication.java:424]: Already covered by `santizeFileName`?

File: container-disc/src/test/java/com/yahoo/container/jdisc/ConfiguredApplicationTest.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.yahoo.container.jdisc;
```
Comment (bjorncs) [inline container-disc/src/test/java/com/yahoo/container/jdisc/ConfiguredApplicationTest.java:1]: Missing copyright header

**Review by bjorncs — APPROVED** [review]

File: container-disc/src/test/java/com/yahoo/container/jdisc/ConfiguredApplicationTest.java (hunk 2 above)
Comment (baldersheim) [reply container-disc/src/test/java/com/yahoo/container/jdisc/ConfiguredApplicationTest.java:2]: Fixed

**Review by bjorncs — APPROVED** [review]

File: container-disc/src/main/java/com/yahoo/container/jdisc/ConfiguredApplication.java (hunk 1 above)
Comment (baldersheim) [reply container-disc/src/main/java/com/yahoo/container/jdisc/ConfiguredApplication.java]: Fixed

**Pull request merged**